### PR TITLE
Implement fetching redirect URLs with token

### PR DESCRIPTION
**Pull request by KillerX**

### CODE REVIEW

File: backend/graph/api/schema.resolvers.go [hunk 1]
```diff
@@ -182,6 +186,59 @@ func (r *queryRootResolver) Export(ctx context.Context, groups []string) (*model
...
+
+	profile := user.GetProfileFromCtx(ginCtx)
+	if profile == nil {
+		return nil, merry.New(
+			"Not authorized",
+			merry.WithUserMessage("you are not authorized for this query"),
+		)
+	}
+
+	redir, err := r.Queries.GetRedirectByCode(ctx, id)
```
Comment (fredrikved) [inline backend/graph/api/schema.resolvers.go:204]: Do we want a cache here perhaps?

Put it through a dataloader and it should be cached, invalid ids as well.

File: migrations/00064_add_redirects.sql [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	"code" varchar(255) NOT NULL DEFAULT NULL::character varying ,
+	CONSTRAINT "redirects_pkey" PRIMARY KEY (id) ,
+	CONSTRAINT "redirects_user_updated_foreign" FOREIGN KEY (user_updated) REFERENCES directus_users(id) ,
+	CONSTRAINT "redirects_user_created_foreign" FOREIGN KEY (user_created) REFERENCES directus_users(id)
+);
+
+ALTER TABLE IF EXISTS "public"."redirects" OWNER TO builder;
+
+GRANT SELECT ON TABLE "public"."redirects" TO api; --WARN: Grant\Revoke privileges to a role can occure in a sql error during execution if role is missing to the target database!
+GRANT INSERT, INSERT, UPDATE, DELETE ON TABLE "public"."redirects" TO directus; --WARN: Grant\Revoke privileges to a role can occure in a sql error during execution if role is missing to the target database!
```
Comment (fredrikved) [inline migrations/00064_add_redirects.sql:26]: Duplicate insert grant? It probably doesn't fail but it's marked as an error here.

**Review by fredrikved — DISMISSED** [review]

File: migrations/00064_add_redirects.sql (hunk 2 above)
Comment (KillerX) [inline migrations/00064_add_redirects.sql:26]:
```suggestion
GRANT INSERT, UPDATE, DELETE ON TABLE "public"."redirects" TO directus; --WARN: Grant\Revoke privileges to a role can occure in a sql error during execution if role is missing to the target database!
```

File: backend/graph/api/schema.resolvers.go (hunk 1 above)
Comment (KillerX) [reply backend/graph/api/schema.resolvers.go:204]: Yes, very good point. This is actually an action that is likely very spiky.

**Pull request merged**